tests/components/amberelectric/test_services.py

```diff
@@ -0,0 +1,206 @@
+"""Test the Amber Service object."""
+
+from collections.abc import Generator
+from datetime import date
+from unittest.mock import Mock, patch
+
+from amberelectric.models.channel import Channel, ChannelType
+from amberelectric.models.site import Site
+from amberelectric.models.site_status import SiteStatus
+import pytest
+
+from homeassistant.components.amberelectric.const import (
+    CONF_SITE_ID,
+    CONF_SITE_NAME,
+    DOMAIN,
+    GET_FORECASTS_SERVICE,
+)
+from homeassistant.components.amberelectric.services import (
+    ATTR_CHANNEL_TYPE,
+    ATTR_SITE_ID,
+)
+from homeassistant.const import CONF_API_TOKEN
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ServiceValidationError
+
+from . import setup_integration
+from .helpers import FORECASTS, GENERAL_ONLY_SITE_ID
+
+from tests.common import MockConfigEntry
+
+
+async def create_amber_config_entry(site_id):
+    """Create an Amber config entry."""
+    return MockConfigEntry(
+        domain=DOMAIN,
+        data={
+            CONF_API_TOKEN: "TOKEN",
+            CONF_SITE_NAME: "home",
+            CONF_SITE_ID: site_id,
+        },
+        entry_id=site_id,
```

Comment:
isn't the site_id the unique id?

Reply:
No, the unqiue id is the concatenation of the site id and the channel type - this setup was wrong previously.